test/system/505-networking-pasta.bats

```diff
@@ -0,0 +1,675 @@
+#!/usr/bin/env bats   -*- bats -*-
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Networking with pasta(1)
+#
+# Copyright (c) 2022 Red Hat GmbH
+# Author: Stefano Brivio <sbrivio@redhat.com>
+
+load helpers
+load helpers.network
+
+function setup() {
+	basic_setup
+	skip_if_no_pasta "pasta not found: install pasta(1) to run these tests"
+
+	RANGE=0		&& TRANSLATED=0
+	DST_PORT=	&& DST_SEQ=
+	PASTA_SPEC=	&& PODMAN_SPEC_PORT=
+	ADDRESS_BOUND=0	&& INTERFACE_BOUND=0
+	FORWARDING=0	&& TRANSFER=0 && BLOCK_SIZE=2k
+
+	# NOTE: Consider the keywords below before picking test names
```

Comment:
Yeouch. No. This is action-at-a-distance, and prone to all sorts of potential errors. Please find a different way to refactor, one in which the tests themselves define the parameters.

Reply:
I'll go back to something similar to my initial version, then. But note that I refactored tests because you seemed to initially indicate they are very long, and we have 48 tests for forwarding, 16 for data transfers. Duplicating the assignments for 5 values would result in approximately 350 additional lines.

While at it, can you be more specific about the potential errors? As I see it, not duplicating assignments for every single test might actually prevent typos. And unfortunately `bats` doesn't really provide a way to create a parameterised test matrix, so I don't see a better way to keep this slim.

Reply:
The first version looks more reasonable and shouldn't cause much duplication, the second one would squash everything into a single test case, which doesn't look very helpful to track down failures.

I don't understand, why would I be the only one who can run tests? Packages for several distributions are available. If missing for a specific distribution, it's rather easy to build (`git clone`, `make`), and I'm also distributing static builds and unofficial packages (x86_64 only). I earlier mentioned in the pull request:

> This pull request introduces a number of tests, **some of which will fail** with the current upstream version of pasta, because I just found a few issues while writing those. Patches are on their [way](https://archives.passt.top/), and they should be available in packages within a few days, but I thought I would submit this early for review anyway.

...all known failures are fixed upstream by now, and currently also in the `updates-testing` versions of Fedora 35, 36, 37 and Rawhide packages (see e.g. https://bodhi.fedoraproject.org/updates/FEDORA-2022-a5ccf10f50). But anyway most tests would work even with the current `stable` version.

Reply:
>the second one would squash everything into a single test case, which doesn't look very helpful to track down failures.

Bingo - very nice observation. That's one of the unfortunate tradeoffs of bats, in particular, not having a test-generator mechanism:
```bash
for ip in... for proto in ... do
        create a new @test with $proto $ip ...
done
...
run each new @test 
```
That would be so sweet! But it's wishful thinking, so the test writer has to make a tradeoff:
* duplication: hard to read, impossible to verify that all permutations are covered
* iteration: clean, easy to read and maintain, especially when new test conditions are added. But, as you say, if one test fails there's no visibility into what other tests fail. This makes it hard to diagnose failures.

It has been our practice to go with the latter. These are system tests, a last-minute sanity check. They are not expected to fail. If they fail, something is very badly broken, and it is appropriate to focus all resources on fixing the problem. And, when these failures do happen, you're right: we have incomplete information. It can be hard. But not as hard as explaining to a customer that oops, no, with all that duplication, we missed one test case, and that's why we broke your workflow, and yes, we'll now spend a lot lot lot of effort to issue a post-release fix.

For the sake of code readability and maintainability, I encourage you to refactor and deduplicate as much as you can, but ultimately it's your call.

Reply:
Done, with the first approach of the two you suggested. I find it less readable than the original, but if we can't depend on test names, so be it. I think I managed to keep duplication to a minimum anyway, by reworking the grouping of options and parts of command lines (especially around socat).